test/dynamo/test_autograd_function.py

```diff
@@ -1076,6 +1076,30 @@ def foo(x):
         foo(torch.randn(2, requires_grad=True))
         self.assertEqual(cnts.frame_count, 1)
 
+    def test_mark_non_differentiable(self):
+        from torch.autograd import Function
+
+        class MyFunction(Function):
+            @staticmethod
+            def forward(ctx, input):
+                output = input > 0
+                ctx.mark_non_differentiable(output)
+                return output
+
+            @staticmethod
+            def backward(ctx, grad_output):
+                return (grad_output * 0).to(torch.double)
+
+        @torch.compile(backend="aot_eager", fullgraph=True)
+        def fn(x):
+            return MyFunction.apply(x)
+
+        x = torch.randn(5, 5, requires_grad=True)
+        mask = fn(x)
+        self.assertFalse(mask.requires_grad)
```

Comment:
I don't know why this works even we don't pass the outputs has been marked as non differentiable into AOT autograd. cc @zou3519

Reply:
Non-floating-point tensors cannot have requires_grad=True. You should test returning a Tensor that is floating point (like output = input.sin(); ctx.mark_non_differentiable(output)

Reply:
If I switch to float tensors, it errors out. I'll update this PR soon.
But this is a unit test that I copied from eager mode, so it seems we need to make that unit test more reasonable as well.
https://github.com/pytorch/pytorch/blob/311af3b9889563e5441fee77bf432edc081f391d/test/test_autograd.py#L3292